### DISCUSSION TITLE
How does one get the MSBuild target to run in diagnostic mode

### DISCUSSION THREAD

**Discussion by danijam** [GitTools/GitVersion#2945] [Q&A] [answered]
I'm sure this is a simple thing but I'm struggling on what to search for.

If I'm running GitVersion locally from the CLI I can using the /diag option to output. diagnostic output that gives more infor on how a version number is generated.

I'm running GitVersion as the MSBuild Target. Can someone help me know how to tell that to run in the diag mode please?

**Comment by asbjornu** [comment, marked as answer]
I'm not using the MSBuild task myself, but I believe [configuring verbosity in MSBuild](https://docs.microsoft.com/en-us/visualstudio/msbuild/obtaining-build-logs-with-msbuild?view=vs-2022) will cause GitVersion to log more information. I don't think you'll be able to replicate the behaviour of `/diag`, though.

**Reply by danijam** [reply]
Thanks @asbjornu. Ah that makes sense, I've tried it but yes unfortunately it doesn't output the INFO statements that you get with /diag. I went down the MSBuild route as that nicely encapsulates the versioning so it works anywhere. I.e local developer gets the same version if they build from the same commit/branch that say a AzDO Pipeline has just built from. 

diag is just useful if there are questions over behaviour nice to be able to just get at that from a pipeline run etc etc.
